modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/renderer/FragmentEntryFragmentRenderer.java

```diff
@@ -347,6 +353,73 @@ private String _renderFragmentEntryLink(
 		return content;
 	}
 
+	// TODO: this method assumes that the imports information comes in a header
+	// in the JS code (because the bundler has put it there). This is only for
+	// the PoC and should be moved to a more suitable place in the final
+	// version (perhaps the fragment.json file or something like that).
+
+	private String _rewriteRequiredModules(long fragmentEntryId, String js) {
+		if (!js.startsWith("//{imports")) {
+			return js;
+		}
+
+		List<String> modules = new ArrayList<>();
+		List<String> variables = new ArrayList<>();
+
+		String[] lines = js.split(StringPool.NEW_LINE);
+
+		for (int i = 1; i < lines.length; i++) {
+			String line = lines[i];
+
+			if (line.equals("//}imports")) {
+				break;
+			}
+
+			line = line.substring(2);
+
+			String[] parts = line.split("\\|");
+
+			variables.add(parts[0]);
+
+			String moduleName = parts[1];
+
+			String packageName = ModuleNameUtil.getPackageName(moduleName);
+
+			// TODO: this is a bit of a hack because we are passing a null
+			// JSPackage owner to JSPackageDependency, but it can be used in
+			// older versions of the portal if we need to as the NPMRegistry
+			// doesn't use it.
+
+			JSPackage jsPackage = _npmRegistry.resolveJSPackageDependency(
+				new JSPackageDependency(null, packageName, parts[2]));
+
+			if (jsPackage == null) {
+				throw new IllegalStateException(
+					StringBundler.concat(
+						"Unable to resolve package version for module ",
+						moduleName, " in fragment entry ", fragmentEntryId));
+			}
+
+			moduleName = StringUtil.replace(
+				moduleName, packageName, jsPackage.getResolvedId());
+
+			modules.add(
+				StringPool.APOSTROPHE + moduleName + StringPool.APOSTROPHE);
+		}
+
+		StringBuilder sb = new StringBuilder();
```

Comment:
Why don't we simply use `ScriptData` like in every other place where we require things rather than writing this from scratch?